includes/Checker/Checks/Plugin_Readme_Check.php

```diff
@@ -42,8 +42,17 @@ public function get_categories() {
 	 * @param array        $files  Array of plugin files.
 	 */
 	protected function check_files( Check_Result $result, array $files ) {
+
+		$plugin_relative_path = $result->plugin()->path();
+		$root_readme          = array(
+			$plugin_relative_path . 'readme.txt',
+			$plugin_relative_path . 'readme.md',
+		);
+
 		// Find the readme file.
-		$readme = self::filter_files_by_regex( $files, '/readme\.(txt|md)$/i' );
+		$readme_list = self::filter_files_by_regex( $files, '/readme\.(txt|md)$/i' );
+
+		$readme = array_intersect( $root_readme, $readme_list );
```

Comment:
Actually, there is a problem here: We have to consider the file names regardless of case, for example, it could also be `README.md`. It would be great to find an alternative solution where we don't have to hardcode all the file names and use an exact string match.

Potentially you could use `array_filter()` to ensure that only readme files are considered which are directly in the plugin root?

Additionally, to your question, we need to add logic here to ignore any `.md` versions _if_ there are `.txt` versions.

Reply:
Thanks @felixarntz for the feedback. In https://github.com/10up/plugin-check/pull/241/commits/3c15ebeff992ca809841dc35e179f29735399d71 I address this. Ready for review.